### PR TITLE
Add TimeZoneInfo APIs remarks

### PR DESCRIPTION
**Pull request by tarekgh**
## Summary

Adding a remark section to the TimeZoneInfo APIs that depend on ICU library.

### CODE REVIEW

**Comment by tarekgh** [comment]
@gewarren could you please help review this? Thanks!

**Comment by tarekgh** [comment]
Looks the change [`b9825e2` (#9161)](https://github.com/dotnet/dotnet-api-docs/pull/9161/commits/b9825e26e7efea38f61624707c0ed7ce359dd94e) not included in the preview build. Maybe the CI finished building before picking up this change?

File: xml/System/TimeZoneInfo.xml [hunk 1]
```diff
@@ -3410,7 +3410,21 @@ This method returns a new <xref:System.TimeZoneInfo> instance for each method ca
         <summary>Tries to convert an IANA time zone ID to a Windows ID.</summary>
         <returns>
           <see langword="true" /> if the ID conversion succeeded, <see langword="false" /> otherwise.</returns>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This API is exclusively supported on .NET 6.0 and later versions. It is also supported when the application utilizes the ICU library, otherwise the API will fail.
```
Comment (gewarren) [inline xml/System/TimeZoneInfo.xml:3418]:
```suggestion
This API is exclusively supported on .NET 6 and later versions, and only when the application utilizes the ICU library.
```

File: xml/System/TimeZoneInfo.xml [hunk 2]
```diff
@@ -3410,7 +3410,21 @@ This method returns a new <xref:System.TimeZoneInfo> instance for each method ca
...
+However, in .NET, the ICU library is not always used by default, particularly in the following scenarios:
+
+- When the application runs on an operating system that does not have the ICU library, and the app does not utilize the [APP-Local ICU](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#app-local-icu) feature.
```
Comment (gewarren) [inline xml/System/TimeZoneInfo.xml:3421]:
```suggestion
- When the application runs on an operating system that does not have the ICU library, and the app does not utilize the [app-local ICU](/dotnet/core/extensions/globalization-icu#app-local-icu) feature.
```

File: xml/System/TimeZoneInfo.xml [hunk 3]
```diff
@@ -3410,7 +3410,21 @@ This method returns a new <xref:System.TimeZoneInfo> instance for each method ca
...
+- When the [Globalization Invariant Mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) or [NLS mode](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#use-nls-instead-of-icu) is enabled on Windows OS.
```
Comment (gewarren) [inline xml/System/TimeZoneInfo.xml:3422]:
```suggestion
- When the [globalization invariant mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) or [NLS mode](/dotnet/core/extensions/globalization-icu#use-nls-instead-of-icu) is enabled on Windows.
```

File: xml/System/TimeZoneInfo.xml [hunk 4]
```diff
@@ -3410,7 +3410,21 @@ This method returns a new <xref:System.TimeZoneInfo> instance for each method ca
...
+
+If the application is running on a Windows OS version that lacks the ICU library, it has the option to opt-in and enable the usage of the ICU library through the [App-Local ICU](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#app-local-icu) feature.
```
Comment (gewarren) [inline xml/System/TimeZoneInfo.xml:3424]:
```suggestion
If the application is running on a Windows OS version that lacks the ICU library, it has the option to opt-in and enable the usage of the ICU library through the [app-local ICU](/dotnet/core/extensions/globalization-icu#app-local-icu) feature.
```

**Review by gewarren — APPROVED** [review]
Couple of small suggestions.

**Pull request merged**